packages/boot/src/controller.booter.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright IBM Corp. 2013,2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {CoreBindings, Application, Booter, BootOptions} from '@loopback/core';
+import {inject} from '@loopback/context';
+import * as glob from 'glob';
+
+/**
+ * ControllerBooter is a class that implements the Booter inferface. The purpose
+ * of this booter is to allow for convention based booting of `Controller`
+ * artifacts for LoopBack 4 Applications.
+ *
+ * It supports the following boot phases: config, discover, boot
+ */
+export class ControllerBooter implements Booter {
+  options: ControllerOptions;
+  projectRoot: string;
+
+  /**
+   *
+   * @param app Application instance
+   * @param bootConfig Config options for boot
+   */
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) public app: Application,
+    @inject(CoreBindings.BOOT_CONFIG) public bootConfig: BootOptions,
+  ) {
+    if (!bootConfig.controllers) bootConfig.controllers = {};
+    this.options = bootConfig.controllers;
+    this.projectRoot = bootConfig.projectRoot;
+  }
+
+  /**
+   * This phase is responsible for configuring this Booter. It converts
+   * options and assigns default values for missing values so ther phases
+   * don't have to perform checks / conversions.
+   */
+  async config() {
+    this.options = Object.assign({}, ControllerDefaults, this.options);
+    if (typeof this.options.dirs === 'string') {
+      this.options.dirs = [this.options.dirs];
+    }
+    if (typeof this.options.extensions === 'string') {
+      this.options.extensions = [this.options.extensions];
+    }
+  }
+
+  /**
+   * This phase is responsible for discovering artifact files based on the
+   * given parameters. Sets options.discovered to an array of discovered
+   * artifact files.
+   */
+  async discover() {
+    const dirs = <string[]>this.options.dirs;
+    const exts = <string[]>this.options.extensions;
+
+    // glob pattern
+    let pattern = `/@(${dirs.join('|')})/`;
+    pattern += this.options.nested ? '**/*' : '*';
+    pattern += `@(${exts.join('|')})`;
+
+    this.options.discovered = glob.sync(pattern, {root: this.projectRoot});
```

Comment:
-1 for using `glob.sync`. Please use `promisify(glob)` and async/await instead.

```ts
import * as glob_ from 'glob';
const glob = promisify(glob_);

this.options.discovered = await glob(pattern, {root: this.projectRoot});
```

Reply:
promisified version of `glob` didn't work for me as it doesn't give me the option to pass an Options object, it just accepts the `pattern` string ...

Reply:
I find that hard to believe because we are already using promisified `glob` with `options` here: https://github.com/strongloop/loopback-next/blob/c951ff1dae43cd503e917e47a2dcd5fc75a3d825/packages/cli/test/clone-example.test.js#L28-L36

Reply:
@virkt25 could you please double-check?

Reply:
Ahh ok. I was using `promisify` from `@loopback/core` which is what was causing issues for me whereas the promisify in CLI uses `const promisify = util.promisify || require('util.promisify/implementation');`. I'll switch to this same version of promisify which should work. 

Thanks for pointing this out! I was running into another issue with `promisify` in creating the "sandboxer" PR which this will solve.

Reply:
I am glad you found a way how to promisify glob!

Is your patch based on the latest `master`? `promisify` in `@loopback/core` is using `equire('util.promisify/implementation')` under the hood too. If there is a bug in our current implementation then we should fix it! See here: https://github.com/strongloop/loopback-next/blob/0df1e0d1affbadf9e9a3a4927048f179ee22763c/packages/core/src/promisify.ts